objects/src/testing/account_builder.rs

```diff
@@ -0,0 +1,115 @@
+use alloc::vec::Vec;
+use core::fmt::Display;
+
+use assembly::Assembler;
+use miden_crypto::merkle::MerkleError;
+use rand::Rng;
+
+use super::account_id::AccountIdBuilder;
+use crate::{
+    accounts::{
+        Account, AccountCode, AccountComponent, AccountStorage, AccountStorageMode, AccountType,
+        AssembledAccountComponent,
+    },
+    assets::{Asset, AssetVault},
+    AccountError, AssetVaultError, Felt, Word, ZERO,
+};
+
+/// Builder for an [`Account`].
+/// The builder allows for a fluent API to construct an account. Each account needs a unique
+/// builder.
+#[derive(Clone)]
+pub struct AccountBuilder<T> {
+    assembler: Assembler,
+    assets: Vec<Asset>,
+    components: Vec<AssembledAccountComponent>,
+    nonce: Felt,
+    account_id_builder: AccountIdBuilder<T>,
+}
```

Comment:
This builder is for testing purposes only, right? We'll probably need another builder (for non-testing purposes) and maybe there is a way to unify the two - but that's probably for a follow-up PR.

Reply:
Yes, that is only for `testing`. I could move it out of `testing` in a separate PR or if that is not possible, make a proper public one. I agree it would be nice to have as part of the public interface.

Reply:
Yep - let's revisit this in a separate PR.